service/frontend/admin/handler_test.go

```diff
@@ -1619,3 +1621,950 @@ func TestGetReplicationMessages(t *testing.T) {
 		})
 	}
 }
+
+func Test_GetDomainReplicationMessages(t *testing.T) {
+	tests := map[string]struct {
+		input          *types.GetDomainReplicationMessagesRequest
+		hcHandlerFunc  func(mock *history.MockClient)
+		drqHandlerFunc func(mock *domain.MockReplicationQueue)
+		wantErr        bool
+	}{
+		"nil request": {
+			input:          nil,
+			hcHandlerFunc:  func(mock *history.MockClient) {},
+			drqHandlerFunc: func(mock *domain.MockReplicationQueue) {},
+			wantErr:        true,
+		},
+		"with default last message ID": {
+			input: &types.GetDomainReplicationMessagesRequest{
+				LastRetrievedMessageID: common.Int64Ptr(-1), // default value
+				ClusterName:            "test-cluster",
+			},
+			hcHandlerFunc: func(mock *history.MockClient) {
+			},
+			drqHandlerFunc: func(mock *domain.MockReplicationQueue) {
+				mock.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{
+					"test-cluster": 1,
+				}, nil)
+				mock.EXPECT().GetReplicationMessages(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, int64(2), nil)
+				mock.EXPECT().UpdateAckLevel(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
```

Comment:
I would suggest not using goock.Any() on write operations like this where you want to ensure that if you were to do a big refactor after this test was landed, that if they were to get zero'd out or were wrong or something that the test would still pass. 

Asserting - particularly in write cases - at least the important values, at both ends is probably worthwhile